Modules/_multiprocessing/posixshmem.c

```diff
@@ -62,6 +62,52 @@ _posixshmem_shm_open_impl(PyObject *module, PyObject *path, int flags,
 }
 #endif /* HAVE_SHM_OPEN */
 
+#ifdef HAVE_SHM_RENAME
+/*[clinic input]
+_posixshmem.shm_rename
+    path_from: unicode
+    path_to: unicode
+    flags: int
+
+Rename a shared memory object.
+
+Remove a shared memory object and relink to another path
+By default, if the destination path already exist, it will be unlinked.
+With the SHM_RENAME_EXCHANGE flag, source and destination paths
+will be exchanged.
+With the SHM_RENAME_NOREPLACE flag, an error will be triggered
+if the destination alredady exists.
+
+[clinic start generated code]*/
+
+static PyObject *
+_posixshmem_shm_rename_impl(PyObject *module, PyObject *path_from,
+                            PyObject *path_to, int flags)
+/*[clinic end generated code: output=a9101f606826ad30 input=e32a44cee3e0326e]*/
+{
+    int rv;
+    int async_err = 0;
+    const char *from = PyUnicode_AsUTF8(path_from);
+    const char *to = PyUnicode_AsUTF8(path_to);
+    if (from == NULL || to == NULL) {
+        return NULL;
+    }
```

Comment:
This function has the same bug as shm_open() and shm_unlink() (see #115886). Please check the names for embedded null characters.